pandas/io/formats/style.py

```diff
@@ -553,11 +576,11 @@ def to_excel(
         startcol: int = 0,
         engine: str | None = None,
         merge_cells: ExcelWriterMergeCells = True,
-        encoding: str | None = None,
```

Comment:
The ordering of these parameters should not change.

Reply:
This function's parameters are out of order when used in the base `to_excel` template docstring (since var `extra_parameters` is substituted at its end). I'll roll back the current change and instead apply a separate docstring template here.

Reply:
Resolution for requested change is in https://github.com/pandas-dev/pandas/pull/59607/commits/175dd3651a3cf7fa9d630f2a12e2edda9dee8438.